### ISSUE TITLE
preload: undefined context in loader if navigation happens before beforeLoad is done

### ISSUE THREAD

**Issue by freshgiammi** [TanStack/router#1886]
### Describe the bug

There's still an issue here when with context when it gets merged with beforeLoad.

```
export const Route = createFileRoute('/about')({
  beforeLoad: async () => {   
    await new Promise((resolve) => setTimeout(resolve, 5000));
    return { someData: 'hello' };
  },
   loader: async ({context}) => {
    await new Promise((resolve) => setTimeout(resolve, 2000));
    console.log(context.someData)
  },
})
```

Supposing `/about` is the lazy route that takes a long time to resolve, if beforeLoad isn't finished before the navigation event happens, the loader gets invoked with an `undefined` context, breaking the route.

This possibly happens because [execution flow is skipped](https://github.com/TanStack/router/blob/main/packages/react-router/src/router.ts#L1772) when a route already in `beforeLoad` fetching state is re-loaded (because the loader gets invoked regardless of the beforeLoad state, for which we have no pending information on reload)

### Your Example Website or App

https://stackblitz.com/edit/github-nkiyw7

### Steps to Reproduce the Bug or Issue

1. Hover `About` link
2. Immediately click to navigate before the 5000ms beforeLoad is done executing
3. Undefined context in loader 🤯 

### Expected behavior

As a user, I expect context to not be undefined

### Platform

- OS: MacOS
- Browser: Chrome
- Version: 1.43.6

**Comment by schiller-manuel** [comment]
this will be fixed by #1907 

reproducer with pre-release package: https://stackblitz.com/edit/github-nkiyw7-ojt8yk?file=src%2Froutes%2Fabout.tsx

**Comment by freshgiammi** [comment]
> this will be fixed by #1907 
> 
> reproducer with pre-release package: https://stackblitz.com/edit/github-nkiyw7-ojt8yk?file=src%2Froutes%2Fabout.tsx

I guess I can close https://github.com/TanStack/router/pull/1891 then?

**Comment by schiller-manuel** [comment]
yes we'll close #1891, but I will take over the test from there, so thanks for implementing that!
